### PR TITLE
Add table `aws_codedeploy_deployment_config` closes #1653

### LINKED ISSUE #1653: Add table aws_codedeploy_deployment_config
**References**
https://github.com/aws/aws-sdk-go-v2/blob/service/codedeploy/v1.16.1/service/codedeploy/api_op_ListDeploymentConfigs.go#L16

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codedeploy_deployment_config

Note: This required for `codedeploy-lambda-allatonce-traffic-shift-disabled` config rule query

### PR DESCRIPTION
**Pull request by madhushreeray30**
# Integration test logs
<details>
  <summary>Logs</summary>

```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_codedeploy_deployment_config []

PRETEST: tests/aws_codedeploy_deployment_config

TEST: tests/aws_codedeploy_deployment_config
Running terraform
data.aws_region.alternate: Reading...
data.aws_partition.current: Reading...
data.aws_region.primary: Reading...
data.aws_caller_identity.current: Reading...
data.aws_region.primary: Read complete after 0s [id=us-east-1]
data.aws_region.alternate: Read complete after 0s [id=us-east-2]
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_caller_identity.current: Read complete after 1s [id=123412341234]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_codedeploy_deployment_config.named_test_resource will be created
  + resource "aws_codedeploy_deployment_config" "named_test_resource" {
      + compute_platform       = "Lambda"
      + deployment_config_id   = (known after apply)
      + deployment_config_name = "turbottest22769"
      + id                     = (known after apply)

      + traffic_routing_config {
          + type = "TimeBasedLinear"

          + time_based_linear {
              + interval   = 10
              + percentage = 10
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id       = "123412341234"
  + compute_platform = "Lambda"
  + resource_id      = (known after apply)
  + resource_name    = "turbottest22769"
aws_codedeploy_deployment_config.named_test_resource: Creating...
aws_codedeploy_deployment_config.named_test_resource: Creation complete after 2s [id=turbottest22769]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 43, in data "null_data_source" "resource":
  43: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "123412341234"
compute_platform = "Lambda"
resource_id = "11111111-2222-3333-4444-555555555555"
resource_name = "turbottest22769"

Running SQL query: test-get-query.sql
[
  {
    "compute_platform": "Lambda",
    "deployment_config_id": "11111111-2222-3333-4444-555555555555",
    "deployment_config_name": "turbottest22769"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "compute_platform": "Lambda",
    "deployment_config_id": "11111111-2222-3333-4444-555555555555",
    "deployment_config_name": "turbottest22769"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "compute_platform": "Lambda",
    "deployment_config_id": "11111111-2222-3333-4444-555555555555",
    "deployment_config_name": "turbottest22769"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "123412341234",
    "title": "turbottest22769"
  }
]
✔ PASSED

POSTTEST: tests/aws_codedeploy_deployment_config

TEARDOWN: tests/aws_codedeploy_deployment_config

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```sql
> select
  arn,
  deployment_config_id,
  deployment_config_name,
  compute_platform,
  create_time,
  region
from
  aws_codedeploy_deployment_config limit 2;
+--------------------------------------------------------------------------------------------------------------+--------------------------------------+--------------------------------------------------+------------------+---------------------------+------------+
| arn                                                                                                          | deployment_config_id                 | deployment_config_name                           | compute_platform | create_time               | region     |
+--------------------------------------------------------------------------------------------------------------+--------------------------------------+--------------------------------------------------+------------------+---------------------------+------------+
| arn:aws:codedeploy:ap-south-1:123412341234:deploymentconfig:CodeDeployDefault.LambdaCanary10Percent30Minutes | 00000000-0000-0000-0000-000000000004 | CodeDeployDefault.LambdaCanary10Percent30Minutes | "Lambda"         | <null>                    | ap-south-1 |
| arn:aws:codedeploy:ap-south-1:123412341234:deploymentconfig:test-conf                                        | 11111111-2222-3333-4444-555555555555 | test-conf                                        | "Server"         | 2022-07-25T18:35:26+05:30 | ap-south-1 |
+--------------------------------------------------------------------------------------------------------------+--------------------------------------+--------------------------------------------------+------------------+---------------------------+------------+
```

```sql
> select
  arn,
  deployment_config_id,
  deployment_config_name,
  compute_platform,
  create_time,
  region
from
  aws_codedeploy_deployment_config
where
  deployment_config_name = 'test-conf';
+-----------------------------------------------------------------------+--------------------------------------+------------------------+------------------+---------------------------+------------+
| arn                                                                   | deployment_config_id                 | deployment_config_name | compute_platform | create_time               | region     |
+-----------------------------------------------------------------------+--------------------------------------+------------------------+------------------+---------------------------+------------+
| arn:aws:codedeploy:ap-south-1:123412341234:deploymentconfig:test-conf | 11111111-2222-3333-4444-555555555555 | test-conf              | "Server"         | 2022-07-25T18:35:26+05:30 | ap-south-1 |
+-----------------------------------------------------------------------+--------------------------------------+------------------------+------------------+---------------------------+------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_codedeploy_deployment_config.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "compute_platform",
+				Description: "The destination platform type for the deployment (Lambda, Server, or ECS).",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getCodeDeployDeploymentConfig,
+			},
+			{
+				Name:        "create_time",
```
Comment (rajlearner17) [inline aws/table_aws_codedeploy_deployment_config.go:52]: @madhushreeray30 this is resulting <null> all the time? What is AWS CLI response?

**Review by rajlearner17 — APPROVED** [review]
LGTM as testing and doc formats

File: docs/tables/aws_codedeploy_deployment_config.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  deployment_config_id,
+  deployment_config_name,
+  compute_platform,
+  create_time,
+  region
+from
+  aws_codedeploy_deployment_config;
+```
+
+### Get total configurations deployed on each platform
```
Comment (karanpopat) [inline docs/tables/aws_codedeploy_deployment_config.md:21]:
```suggestion
### Get the total configurations for each compute platform
```

File: docs/tables/aws_codedeploy_deployment_config.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+select
+  count(arn) as configuration_count,
+  compute_platform
+from
+  aws_codedeploy_deployment_config
+group by
+  compute_platform;
+```
+
+### List the user defined configurations
```
Comment (karanpopat) [inline docs/tables/aws_codedeploy_deployment_config.md:33]:
```suggestion
### List the user managed deployment configurations
```

**Review by karanpopat — COMMENTED** [review]
@madhushreeray30 Please check comments and make changes where necessary

**Review by karanpopat — APPROVED** [review]
LGTM

File: aws/table_aws_codedeploy_deployment_config.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("deployment_config_name"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"DeploymentConfigDoesNotExistException"}),
+			},
+			Hydrate: getCodeDeployDeploymentConfig,
+		},
+		List: &plugin.ListConfig{
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"DeploymentConfigDoesNotExistException"}),
```
Comment (misraved) [inline aws/table_aws_codedeploy_deployment_config.go:32]: Do we get this error in the List API call as well?

File: aws/table_aws_codedeploy_deployment_config.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+			},
+			{
+				Name:        "create_time",
+				Description: "The time at which the deployment configuration was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getCodeDeployDeploymentConfig,
+			},
+			{
+				Name:        "deployment_config_id",
+				Description: "The deployment configuration id.",
```
Comment (misraved) [inline aws/table_aws_codedeploy_deployment_config.go:53]:
```suggestion
				Description: "The deployment configuration ID.",
```

File: docs/tables/aws_codedeploy_deployment_config.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
+# Table: aws_codedeploy_app
```
Comment (misraved) [inline docs/tables/aws_codedeploy_deployment_config.md:1]:
```suggestion
# Table: aws_codedeploy_deployment_config
```

File: docs/tables/aws_codedeploy_deployment_config.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  deployment_config_id,
+  deployment_config_name,
+  compute_platform,
+  create_time,
+  region
+from
+  aws_codedeploy_deployment_config;
+```
+
+### Get the total configurations for each compute platform
```
Comment (misraved) [inline docs/tables/aws_codedeploy_deployment_config.md:21]:
```suggestion
### Get the configuration count for each compute platform
```

**Review by misraved — COMMENTED** [review]
@madhushreeray30 please take a look at the review comments. Thanks!!

**Pull request merged**